File: LFAutomation/Python/add_remove_device.py
# Import the .NET class library
import clr

# Import python sys module
import sys

# Import os module
import os

# Import System.IO for saving and opening files
from System.IO import *

# Import c compatible List and String
from System import String
from System.Collections.Generic import List

# Add needed dll references
sys.path.append(os.environ['LIGHTFIELD_ROOT'])
sys.path.append(os.environ['LIGHTFIELD_ROOT']+"\\AddInViews")
clr.AddReference('PrincetonInstruments.LightFieldViewV5')
clr.AddReference('PrincetonInstruments.LightField.AutomationV5')
clr.AddReference('PrincetonInstruments.LightFieldAddInSupportServices')

# PI imports
from PrincetonInstruments.LightField.Automation import Automation
    

def add_available_devices():
    # Add first available device and return
    for device in experiment.AvailableDevices:
        print("\n\tAdding Device...")
        experiment.Add(device)
        return device

# create a C# compatible List of type String object
list1 = List[String]()

# add the command line option for an empty experiment
list1.Add("/empty")

# Create the LightField Application (true for visible)
auto = Automation(True, List[String](list1))

# Get experiment object
experiment = auto.LightFieldApplication.Experiment


# Check for device and inform user if one is needed
if (experiment.AvailableDevices.Count == 0):
    print("Device not found. Please add a device and try again.")
else:
    device = add_available_devices()
    
    print("\n\tRemoving Device...")
    
    # Removes the current device
    experiment.Remove(device)


# Result: LightField will add and remove a device.
'''
	Adding Device...

	Removing Device...
'''

File: LFAutomation/MATLAB/add_remove_device.m
clear all

Setup_LightField_Environment;
instance1=lfm(true);

exp=instance1.application.Experiment;

if exp.AvailableDevices.Count == 0 && exp.ExperimentDevices.Count == 0
    fprintf('Device not found. Please add a device and try again.\n');
else
    devAdded=add_available_devices(exp,exp.AvailableDevices.Count);
    remove_added_devices(exp,devAdded);
end

instance1.close();

function devices = add_available_devices(experiment,count)
    for i=1:count
        fprintf('\tAdding Device...\n');
        experiment.Add(experiment.AvailableDevices.Item(i-1));
    end
    devices=experiment.ExperimentDevices.Count;
end

function remove_added_devices(experiment,count)
    for i=1:count
        fprintf('\tRemoving Device...\n');
        experiment.Remove(experiment.ExperimentDevices.Item(i-1));
    end
end
